doc/march_packages/march_data_collector.rst

```diff
@@ -0,0 +1,133 @@
+This is then published as a marker on ``/march/com_marker``.
+The calculation is based on an implementation from `Hamburg Bit-Bots <https://github.com/bit-bots>`_.
+The marker can also be visualized in :ref:`rviz <robot-model-label>`. The tf frames and CoM can be combined to calculate the capture point.
+This is the point where one should place their foot in order to return to a stable standing position.
+This calculation is done for both feet and published as a marker on ``/march/cp_marker_ankle_plate_{left|right}``
+The center of mass and capture point are used for (research on) balancing the exoskeleton.
+
+Event Stream Processing
+-----------------------
+At March we use the `Event Stream Processing` engine from `SAS <https://www.sas.com/nl_nl/home.html>`_.
+Here we will just describe the integration in the `march_data_collector`, but more information on `ESP` and how we use it can be found at :ref:`event stream processing <event-stream-processing-label>`.
+The engine allows for real-time data analysis and processing. With the installation also comes a module to inject events, which is used in this package.
+The :march:`esp_adapter.py <march_data_collector/src/march_data_collector/esp_adapter.py>` functions as the interface between our software and ESP. This ROS node subscribes to
+ROS topics containing interesting data. It converts the ROS messages into csv strings and injects the data into the correct window of the ESP engine.
+Injecting data can only happen into a source window.
+
+
+Adding a publisher into `ESP`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+1.
+    Add a source window to the :march:`model <march_data_collector/esp_models/march.xml>`.
+
+2.
+    Create a callback function that takes a ROS message and calls the ``send_to_esp`` function with the msgs as csv
+    string in :march:`esp_adapter.py <march_data_collector/src/march_data_collector/esp_adapter.py>`.
+
+3.
+    In the same file add a call to ``configure_source`` to the ``__init__``.
+
+Pressure Soles
+--------------
+The `march_data_collector` interfaces between the Moticon pressure soles and the exoskeleton software.
+It  sends data to and receives data from the desktop software that came with the pressure soles.
+This data is send over `UDP <https://nl.wikipedia.org/wiki/User_Datagram_Protocol>`_.
+More information on interacting with the Moticon software can be found `here <https://www.moticon.de/doc/science_desktop_software/record/udp/>`_.
+The package assumes that the filter used in the Moticon software is ``force pressure cop``.
+More information on using the pressure soles can be found at :ref:`using the pressure soles <using-the-pressure-soles-label>`.
+
+ROS API
+-------
+
+Nodes
+^^^^^
+*march_data_collector_node* - Responsible for processing IMU data and calculating the acceleration and jerk.
+
+*com_calculator* - Responsible for calculating the center of mass.
+
+*cp_calculator* - Responsible for calculating the capture points.
```

Comment:
These are technically not separate ROS nodes, but are included in the `march_data_collector_node`.